packages/documents/lib/meta.js

```diff
@@ -3,53 +3,78 @@ const visit = require('unist-util-visit')
 const { metaFieldResolver } = require('./resolve')
 
 /**
- * If not available already, assign credits to doc.content.meta.
+ * Obtain credits from either {doc.content.children} or {doc.meta}.
  *
  * @param  {Object} doc An MDAST tree
- * @return {Object}     Returns maybe altered {doc}
+ * @return {Array}      MDAST children
  */
-const ensureCredits = doc => {
-  if (!doc.content.meta.credits) {
+const getCredits = doc => {
+  // If {doc.content} is available, always obtain credits from it.
+  if (doc.content && doc.content.children) {
+    let credits = []
+
     visit(doc.content, 'zone', node => {
       if (node.identifier === 'TITLE') {
         const paragraphs = node.children
           .filter(child => child.type === 'paragraph')
         if (paragraphs.length >= 2) {
-          Object.assign(
-            doc.content.meta,
-            { credits: paragraphs[paragraphs.length - 1].children }
-          )
+          credits = paragraphs[paragraphs.length - 1].children
         }
       }
     })
+
+    return credits
   }
 
-  return doc
+  // Due to perfomance considerations a {doc} might come in without
+  // {doc.content}, or only {doc.content.meta} without credits is provided (to
+  // resolve). In such a case, we look for {doc.meta.credits} to obtain credits.
+  return (doc.meta && doc.meta.credits) || []
 }
 
+/**
+ * Builds and an audioSource object from {doc.content.meta} for use in meta.
+ *
+ * @param  {Object}      doc An MDAST tree
+ * @return {Object|null}     e.g. { mp3: true, aac: null, ogg: null }
+ */
+const getAudioSource = doc => {
```

Comment:
```suggestion
const getAudioSource = doc => {
  if (!doc.content && doc.meta && doc.meta.audioSource) {
    return doc.meta.audioSource
  }
```
This might help for the upcoming feature of showing an audio icon in the feed/search result.